src/lib.rs

```diff
@@ -311,4 +311,36 @@ mod tests {
         };
         assert_eq!(x, 3);
     }
+
+    #[test]
+    fn magic_deref() {
+        fn f(y: &'static str) -> (&'static str, bool) {
```

Comment:
This can be simplified:

```rust
#[test]
fn magic_deref() {
    struct Wrapped<T>(T);
    let s = "hello";
    if_chain! {
        then { Wrapped(s) }
        else { Wrapped(&s) }
    };
}
```